File: plan_vision_backend/src/project-template/dto/project-template.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { Expose } from 'class-transformer';

export class ProjectTemplateDto {
  @ApiProperty()
  @Expose()
  id: string; // ✅ Changed to string

  @ApiProperty()
  @Expose()
  title: string;

  @ApiProperty({ required: false })
  @Expose()
  description?: string;

  @ApiProperty()
  @Expose()
  thumbnailUrl: string;

  @ApiProperty()
  @Expose()
  sampleImageUrls: string[];

  @ApiProperty()
  @Expose()
  defaultImageTypeId: string;
}

File: plan_vision/plan_vision/src/features/projects/data/DTOs/ProjectTemplateDTO.swift
//
//  ProjectTemplateDTO.swift
//  plan_vision
//
//  Created by Kartik Garasia on 2025-11-22.
//

import Foundation

struct ProjectTemplateDTO: Codable {
    let id: String
    let title: String
    let description: String?
    let thumbnailUrl: String
    let originalThumbnailUrl: String?
    let generatedThumbnailUrl: String?
    let sampleImageUrls: [String]
    let defaultImageTypeId: String
}
